tests/unit_tests/test_ladder_service.py

```diff
@@ -827,14 +827,10 @@ async def test_write_rating_progress_message_2(
 
     ladder_service.write_rating_progress(player, RatingType.LADDER_1V1)
 
-    player.write_message.assert_called_once_with({
-        "command": "notice",
-        "style": "info",
-        "text": (
-            "The system is still learning you.<b><br><br>"
-            "The learning phase is 40% complete<b>"
-        )
-    })
+    player.write_message.assert_called_once()
+    assert player.write_message.call_args[0][0].get("command") == "notice"
```

Comment:
This didn't match what i expected from the documentation for a non-coroutine mock:
https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.call_args

In particular `player.write_message.call_args.args` and `player.write_message.call_args.kwargs` are empty. So I had to experiment a bit and this is what turned out to work, but there might be a slicker way to access it.

Reply:
Why not keep `assert_called_once_with`?. 

I think call_args is a list of NamedTuple's or something like that. `call_args[0]` gives you the args for the first call and `call_args[0][0]` gives you the first arg of the first call.

Reply:
I thought for `assert_called_once_with` you have to provide the full argument list and that's what I wanted to change here so the test doesn't depend on the exact wording of the message anymore.

Reply:
Right